src/g2_unpack2.c

```diff
@@ -48,34 +49,33 @@ g2_unpack2(unsigned char *cgrib, g2int *iofst, g2int *lencsec2,
     *lencsec2 = lensec - 5;
     gbit(cgrib, &isecnum, *iofst, 8);         /* Get Section Number */
     *iofst = *iofst + 8;
-    ipos = (*iofst/8);
+    ipos = *iofst / 8;
 
     if ( isecnum != 2 )
     {
         ierr = 2;
         *lencsec2 = 0;
         fprintf(stderr, "g2_unpack2: Not Section 2 data.\n");
-        return(ierr);
+        return ierr;
     }
 
     if (*lencsec2 == 0)
     {
         ierr = 0;
-        return(ierr);
+        return ierr;
     }
 
     if (!(*csec2 = malloc(*lencsec2 + 1)))
     {
         ierr = 6;
         *lencsec2 = 0;
-        return(ierr);
+        return ierr;
     }
 
-    /*printf(" SAGIPO %d \n", (int)ipos);*/
     for (j = 0; j < *lencsec2; j++)
-        *(*csec2 + j) = cgrib[ipos + j];
+        (*csec2)[j] = cgrib[ipos + j];
```

Comment:
Trying to wrap my head around what's happening here. All looks good otherwise.

Reply:
It's explained in #183.